### ISSUE TITLE
Problems with the new version of Mikrotik

### ISSUE THREAD

**Issue by SerhiiZahuba** [cBackup/main#77]
Starting with firmware 6.49 Mikrotik, backup does not work, on all devices up to version 6.49 - everything works well.

Templates

export
quit

![image](https://user-images.githubusercontent.com/36307887/159288600-d5d48463-7fde-456e-850a-95fae2285c3b.png)

**Comment by guardeatingapple** [comment]
hi, did you solve the problem?

**Comment by guardeatingapple** [comment]
maybe you have the wrong prompt string, for mikrotik it's ">"

**Comment by SerhiiZahuba** [comment]
No. The problem is only if the firmware of the router is higher than 6.49, if the firmware is < 6.49 there are no problems.

**Comment by SerhiiZahuba** [comment]
@launcherx can you help ?

**Comment by technetius** [comment]
There is problem with Java Worker, because of updated SSH shell in new firmware
New output looks like this
```
[admin@MikroTik] >
[admin@MikroTik] > [Ke
[admin@MikroTik] > e[Kx
[admin@MikroTik] > ex[Kp
[admin@MikroTik] > exp[Ko
[admin@MikroTik] > expo[Kr
[admin@MikroTik] > expor[Kt
[admin@MikroTik] > export[K

===== CONFIG =====

[admin@MikroTik] >
```
When expect4j is trying to search for command execution end, it matches first occurrence of `[admin@MikroTik] >`  and exits.

Without updating and recompiling java worker new firmware will not work

**Comment by g-philipp** [comment]
any Updates here?

**Comment by Vaskata84** [comment]
Hello, this is how things work for me My version is 7.8. You can see the screenshots
https://ibb.co/5cQSdy2
https://ibb.co/5jQPYwF
https://ibb.co/bPVdx8t

**Comment by romed77** [comment]
> Hello, this is how things work for me My version is 7.8. You can see the screenshots https://ibb.co/5cQSdy2 https://ibb.co/5jQPYwF https://ibb.co/bPVdx8t

I have tested, this solution works.
Thank you!

**Comment by Guilhermegfssilva** [comment]
hello, with version 7.10.2 this solution did not work.

Returns only:

##CONFIG##



##ENDOF_CONFIG##

Does anyone have any tips/suggestions?

**Comment by Guilhermegfssilva** [comment]
I was able to apply it like this:

https://prnt.sc/OCycY0dAbCkN

**Comment by Vaskata84** [comment]
> I was able to apply it like this:
> 
> https://prnt.sc/OCycY0dAbCkN

does it work like that with telnet?

**Comment by Vaskata84** [comment]
Task backup, node 1: timeout. Telnet response expect failed. Check device auth sequence and credentials. Schedule id: 2 Task name: backup Node id: 1 Worker id: 1 “someone tell me how to fix this"

**Comment by Guilhermegfssilva** [comment]
hello,

I didn't use telnet, only ssh.

**Comment by Vaskata84** [comment]
> 

send me the whole configuration with pictures how you set it up can you ?

**Comment by Vaskata84** [comment]
> hello,
> 
> I didn't use telnet, only ssh.

send me the whole configuration with pictures how you set it up can you ?

**Comment by fabriciomatrix** [comment]
@Guilhermegfssilva @romed77 @Vaskata84

Hello everybody! 

Do any of you still use cbackup for Mikrotik routers?
The system, despite being dead for a long time, is still very good. I haven't found a suitable, simple and functional program yet.
I'm still experiencing this backup error with Mikrotik routers, but I keep trying

![image](https://github.com/cBackup/main/assets/86535948/debfdf72-7ab5-40a4-90a1-9fd3e9101e2f)


Thanks!

**Comment by Guilhermegfssilva** [comment]
Olá @fabriciomatrix 

Seguimos com problemas, mas dentre as alternativas, o comando "export compact" foi o que trouxe a configuração. 

Você encontrou alguma solução?

**Comment by fabricio-gerenciatec** [comment]
Infelzimente sigo com o problema e nenhuma das alternativas funcionou com o v6.49.x ou 7.x do routerOS

**Comment by fabriciomatrix** [comment]
Infelizmente sigo com o problema, mesmo testado todas as possibilidas com router OS v6.49.x quanto v7.x

**Comment by Guilhermegfssilva** [comment]
@fabriciomatrix consegui fazer funcionar desta forma:

https://prnt.sc/6jj87d1vwoEm

Experimenta replicar igual e testar com versões 6.49 que apresentavam problema.

Na versão 7 que testei também funcionou, 7.10.2.

**Comment by iampikus** [comment]
> > Hello, this is how things work for me My version is 7.8. You can see the screenshots https://ibb.co/5cQSdy2 https://ibb.co/5jQPYwF https://ibb.co/bPVdx8t
> 
> I have tested, this solution works. Thank you!

hello! The link is dead, can you repost your solution?
Thanks

**Comment by fabriciomatrix** [comment]
> @fabriciomatrix consegui fazer funcionar desta forma:
> 
> https://prnt.sc/6jj87d1vwoEm
> 
> Experimenta replicar igual e testar com versões 6.49 que apresentavam problema.
> 
> Na versão 7 que testei também funcionou, 7.10.2.

Infelizmente não funcionou para im. Vou tentar pegar um backup que tenho do zero, já que não consigo mais instalar via repositorio

**Comment by renatoornelas** [comment]
You can add +cte to login credential and it should work (please see https://help.mikrotik.com/docs/spaces/ROS/pages/328134/Command+Line+Interface)

**Comment by slaventiy2007-wq** [comment]
[cbackup.jar.txt](https://github.com/user-attachments/files/24792272/cbackup.jar.txt)

With the author’s permission, I rebuilt the worker to use backup based on /system/ssh-exec.
Posting it below.

Instructions:

Rename cbackup.jar.txt to cbackup.jar

service cbackup stop

rm -f /opt/cbackup/bin/cbackup.jar

Place the downloaded file into /opt/cbackup/bin/

chmod 755 cbackup.jar

chown apache:apache cbackup.jar

service cbackup start

In Device auth templates for MikroTik, set it to >.

Profit.

**Comment by Vaskata84** [comment]
Are you interested in a backup system for multiple vendors?

**Comment by slaventiy2007-wq** [comment]
> Are you interested in a backup system for multiple vendors?

We’ve been using cbackup for over 5 years and everything suits us. The only issue was that MikroTik backups stopped working, but after a couple of weeks they were fixed again. So it’s not really of interest to us right now.

**Comment by Vaskata84** [comment]
Are you interested in a custom network monitor solution?
